Sources/SQLiteNIO/SQLiteConnection+Hooks.swift

```diff
@@ -0,0 +1,1098 @@
+    /// - Returns: The return value of the body block.
+    /// - Throws: Any error thrown by the body block.
+    public func withAuthorizerObserver<T>(
+        _ callback: @escaping SQLiteAuthorizerObserver,
+        body: () async throws -> T
+    ) async throws -> T {
+        let token = try await addAuthorizerObserver(lifetime: .scoped, callback)
+        defer { token.cancel() }
+        return try await body()
+    }
+
+    /// Execute a block with a temporary rollback observer (synchronous version).
+    ///
+    /// The observer is automatically removed when the block completes.
+    ///
+    /// - Parameters:
+    ///   - callback: The observer callback to register temporarily.
+    ///   - body: The block to execute with the observer active.
+    /// - Returns: The return value of the body block.
+    /// - Throws: Any error thrown by the body block.
+    public func withRollbackObserver<T>(
+        _ callback: @escaping SQLiteRollbackHookCallback,
+        body: () throws -> T
+    ) rethrows -> T {
+        let token = addRollbackObserver(lifetime: .scoped, callback)
+        defer { token.cancel() }
+        return try body()
+    }
+
+    /// Execute a block with a temporary rollback observer (async version).
+    ///
+    /// The observer is automatically removed when the block completes.
+    ///
+    /// - Parameters:
+    ///   - callback: The observer callback to register temporarily.
+    ///   - body: The block to execute with the observer active.
+    /// - Returns: The return value of the body block.
+    /// - Throws: Any error thrown by the body block.
+    public func withRollbackObserver<T>(
+        _ callback: @escaping SQLiteRollbackHookCallback,
+        body: () async throws -> T
+    ) async throws -> T {
+        let token = try await addRollbackObserver(lifetime: .scoped, callback)
+        defer { token.cancel() }
+        return try await body()
+    }
+}
+
+// MARK: - Token-Based Observers
+
+/// Token-Based Observer APIs
+///
+/// - Tag: TokenObservers
+extension SQLiteConnection {
+    /// Register an observer for the SQLite *update* hook (row-level DML).
+    ///
+    /// Fired whenever a row is inserted, updated, or deleted. Multiple observers
+    /// can be registered on the same connection.
+    ///
+    /// ```swift
+    /// let token = connection.addUpdateObserver(lifetime: .pinned) { [weak self] event in
+    ///     self?.logger.info("\(event.table) row \(event.rowID) was \(event.operation)")
+    /// }
+    /// // Clean up the observer when no longer needed:
+    /// token.cancel()
+    /// ```
+    ///
+    /// - Important: Registration is safe from any thread. Callbacks are invoked on SQLite's
+    ///   internal thread; hop to your own actor or event loop as needed.
+    /// - Parameter lifetime: The observer lifetime behavior. See ``ObserverLifetime`` for details. Use `.scoped` for automatic cleanup on token deallocation, or `.pinned` for explicit cleanup only.
```

Comment:
```suggestion
    /// > Important: Registration is /safe from any thread. Callbacks are invoked on SQLite's
    ///   internal thread; hop to your own actor or event loop as needed.
    ///
    /// - Parameter lifetime: The observer lifetime behavior. See ``SQLiteObserverLifetime`` for details.
    ///    Use ``SQLiteObserverLifetime/scoped`` for automatic cleanup on token deallocation, or
    ///    ``SQLiteObserverLifetime/pinned`` for explicit cleanup only.
```

Same for the rest of these methods.